app/src/main/java/org/gnucash/android/export/csv/CsvTransactionsExporter.java

```diff
@@ -0,0 +1,177 @@
+        super(params, null);
+        mCsvSeparator = params.getCsvSeparator();
+        LOG_TAG = "GncXmlExporter";
+    }
+
+    /**
+     * Overloaded constructor.
+     * Creates an exporter with an already open database instance.
+     * @param params Parameters for the export
+     * @param db SQLite database
+     */
+    public CsvTransactionsExporter(ExportParams params, SQLiteDatabase db) {
+        super(params, db);
+        mCsvSeparator = params.getCsvSeparator();
+        LOG_TAG = "GncXmlExporter";
+    }
+
+    @Override
+    public List<String> generateExport() throws ExporterException {
+        OutputStreamWriter writerStream = null;
+        CsvWriter writer = null;
+        String outputFile = getExportCacheFilePath();
+        try {
+            FileOutputStream fileOutputStream = new FileOutputStream(outputFile);
+            BufferedOutputStream bufferedOutputStream = new BufferedOutputStream(fileOutputStream);
+            writerStream = new OutputStreamWriter(bufferedOutputStream);
+            writer = new CsvWriter(writerStream);
+            generateExport(writer);
+        } catch (IOException ex){
+            Crashlytics.log("Error exporting CSV");
+            Crashlytics.logException(ex);
+        } finally {
+            if (writerStream != null) {
+                try {
+                    writerStream.close();
+                } catch (IOException e) {
+                    throw new ExporterException(mExportParams, e);
+                }
+            }
+        }
+
+        List<String> exportedFiles = new ArrayList<>();
+        exportedFiles.add(outputFile);
+
+        return exportedFiles;
+    }
+
+    public void generateExport(final CsvWriter writer) throws ExporterException {
+        try {
+            String separator = mCsvSeparator + "";
+            List<String> names = new ArrayList<String>();
+            names.add("Date");
+            names.add("Account name");
+            names.add("Number");
+            names.add("Description");
+            names.add("Notes");
+            names.add("Memo");
+            names.add("Category");
+            names.add("Type");
+            names.add("Action");
+            names.add("Reconcile");
+            names.add("To With Sym");
+            names.add("From With Sym");
+            names.add("To Num.");
+            names.add("From Num.");
+            names.add("To Rate/Price");
+            names.add("From Rate/Price");
+
+            List<Transaction> transactions = mTransactionsDbAdapter.getAllTransactions();
+
+            for(int i = 0; i < names.size(); i++) {
+                writer.write(names.get(i) + separator);
+            }
+            writer.write("\n");
+            for(int i = 0; i < transactions.size(); i++) {
+                Transaction transaction = transactions.get(i);
+                List<Split> splits = transaction.getSplits();
+                for (int j = 0; j < splits.size()/2; j++) {
```

Comment:
This loops depends on the ordering of the splits in the list. Even if it might be the case at the moment, I would prefer it we did not depend on it. The logic here would change anyway when switching to cursor-based iteration. When modifying it, please do not rely on the ordering of the splits.

Reply:
I'll try to fix it in 5-10 days